lib/toxiproxy.rb

```diff
@@ -211,6 +211,23 @@ def toxics
 
   private
 
+  def self.http_request(request)
+    ensure_webmock_whitelists_toxiproxy if defined? WebMock
+    http.request(request)
+  end
+
+  def http_request(request)
+    self.class.http_request(request)
+  end
+
+  def self.ensure_webmock_whitelists_toxiproxy
+    endpoint = "#{uri.host}:#{uri.port}"
+    WebMock::Config.instance.allow ||= []
```

Comment:
I thought about this for a little while. Unless the overhead proves to be a problem, I believe this is the best solution for the same reasons you mention.